NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/hitdetection/LagCompensator.java

```diff
@@ -0,0 +1,79 @@
+package dev.cobblesword.nachospigot.hitdetection;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import me.elier.nachospigot.config.NachoConfig;
+import org.apache.commons.lang3.tuple.Pair;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.util.Vector;
+
+import java.util.*;
+
+public class LagCompensator {
+
+    /*
+        Lag Compensator based on https://github.com/Islandscout/ServerSideHitDetection
+        Adjusts hit detection to be more fair for players with worse ping.
+     */
+
+    private final ListMultimap<UUID, Pair<Location, Long>> locationTimes = ArrayListMultimap.create();
+    private final int historySize = 40;
+    private final int pingOffset = 92;
+    private final int timeResolution = 30;
+
+    public Location getHistoryLocation(Player player) {
```

Comment:
Might be worth adding a comment here explaining how it works, like "Gets the closest location to the attackers ping" or something.

Reply:
This made me realize after one of the changes it broke the "compensator". This method used to have another parameter used to specify the ping of the attacker, and it's now instead using the ping of the victim, so that's quite useless. I'm working on fixing this atm